nodejs/week2/app.js

```diff
@@ -0,0 +1,64 @@
+const express = require('express');
+const data = require('./documents.json');
+const app = express();
+const port = process.env.PORT || 3000;
+
+// Support parsing JSON requests
+app.use(express.json());
+
+app.get('/', (req, res) => {
+    res.send('This is a search engine');
+});
+
+app.get('/search', (req, res) => {
+    const queryValue = req.query.q;
+    if (queryValue) {
+        const matchingQuerry = data.filter(item => {
+            if (new RegExp(queryValue, 'i').test(Object.values(item))) {
+                return item;
+            }
+        });
+
+        res.status(200).json(matchingQuerry);
+    } else {
+        res.json(data);
+    }
+});
+
+app.get('/documents/:id', (req, res) => {
+    const id = req.params.id;
+
+    const matchingId = data.filter(item => item.id === parseInt(id));
+
+    if (!matchingId.length) {
+        res.status(404).json('error: Not Found');
+    } else {
+        res.send(matchingId);
+    }
```

Comment:
The name `matchingId `says it is an id, not the document. You can call it `documentWithId ` or something like that.
Variable names should descriptive.

Reply:
It was fixed.